lib/qtermwidget.h

```diff
@@ -62,56 +54,57 @@ class QTERMWIDGET_EXPORT QTermWidget : public QWidget {
     QSize sizeHint() const override;
 
     // expose TerminalDisplay::TerminalSizeHint, setTerminalSizeHint
-    void setTerminalSizeHint(bool enabled);
-    bool terminalSizeHint();
+    void setTerminalSizeHint(bool enabled) override;
+    bool terminalSizeHint() override;
 
     //start shell program if it was not started in constructor
-    void startShellProgram();
+    void startShellProgram() override;
 
     /**
      * Start terminal teletype as is
      * and redirect data for external recipient.
      * It can be used for display and control a remote terminal.
      */
-    void startTerminalTeletype();
+    void startTerminalTeletype() override;
 
-    int getShellPID();
+    int getShellPID() override;
 
-    void changeDir(const QString & dir);
+    void changeDir(const QString & dir) override;
 
     //look-n-feel, if you don`t like defaults
 
     //  Terminal font
     // Default is application font with family Monospace, size 10
     // Beware of a performance penalty and display/alignment issues when using a proportional font.
-    void setTerminalFont(const QFont & font);
-    QFont getTerminalFont();
-    void setTerminalOpacity(qreal level);
-    void setTerminalBackgroundImage(const QString& backgroundImage);
-    void setTerminalBackgroundMode(int mode);
+    void setTerminalFont(const QFont & font) override;
+    QFont getTerminalFont() override;
+    void setTerminalOpacity(qreal level) override;
+    void setTerminalBackgroundImage(const QString& backgroundImage) override;
+    void setTerminalBackgroundMode(int mode) override;
 
     //environment
-    void setEnvironment(const QStringList & environment);
+    void setEnvironment(const QStringList & environment) override;
 
     //  Shell program, default is /bin/bash
-    void setShellProgram(const QString & program);
+    void setShellProgram(const QString & program) override;
 
     //working directory
-    void setWorkingDirectory(const QString & dir);
-    QString workingDirectory();
+    void setWorkingDirectory(const QString & dir) override;
+    QString workingDirectory() override;
 
     // Shell program args, default is none
-    void setArgs(const QStringList & args);
+    void setArgs(const QStringList & args) override;
 
     //Text codec, default is UTF-8
-    void setTextCodec(QTextCodec * codec);
+    void setTextCodec(QTextCodec * codec) override;
 
     /** @brief Sets the color scheme, default is white on black
      *
      * @param[in] name The name of the color scheme, either returned from
      * availableColorSchemes() or a full path to a color scheme.
      */
-    void setColorScheme(const QString & name);
+    void setColorScheme(const QString & name) override;
+    QStringList getAvailableColorSchemes() override;
```

Comment:
It's kind of strange to have to member functions, one static and one non-staic, with the same functionality. Are Qt plugins incompatible with static functions? If functions with similar names are inevitable, please add comments for the reason. [Other static functions](https://github.com/lxqt/qtermwidget/#static-public-members) may need changes as well. Also, please update README for API changes.

Reply:
The PyQt binding https://github.com/lxqt/qtermwidget/blob/master/pyqt/sip/qtermwidget.sip should be updated for API changes as well. It's not mentioned in https://raw.githubusercontent.com/lxqt/qtermwidget/master/.github/PULL_REQUEST_TEMPLATE.md, though.

Reply:
Hi @yan12125 ! I've applied all the changes to the PR now :)

The reason for having this method _also_ as not static is exactly the one you said: for a plugin it's impossible to access a static method because the "entry" point for the plugin is the interface, so all methods must be pure virtual.